src/components/MutateBookmark/MutateBookmark.tsx

```diff
@@ -0,0 +1,201 @@
+import { Button, H0, Stack } from "@deskpro/app-sdk";
+import { useEffect, useState } from "react";
+import { useForm } from "react-hook-form";
+import { v4 as uuidV4 } from "uuid";
+import { useSettingsUtilities } from "../../hooks/useSettingsUtilities";
+import { IBookmark } from "../../types/bookmarks";
+import { DropdownSelect } from "../DropdownSelect/DropdownSelect";
+import { InputWithTitle } from "../InputWithTitle/InputWithTitle";
+type Props = {
+  type: "Add" | "Edit";
+  objectId?: string;
+  folderName?: string;
+  objectName: string;
+  setPageSettings: (
+    obj: {
+      type: "Add" | "Edit";
+      objectName: "Bookmark" | "Folder";
+      objectId?: string;
+    } | null
+  ) => void;
+};
+
+type InputType = {
+  Name: string;
+  URL: string;
+  Description: string;
+  ParentFolder: string | null;
+  Id: string;
+};
+
+export const MutateBookmark = ({
+  type,
+  objectId,
+  objectName,
+  setPageSettings,
+}: Props) => {
+  const bookmarkUtilities = useSettingsUtilities();
+  const [gotParentFolder, setGotParentFolder] = useState<boolean>(false);
+  const [parentFolders, setParentFolders] = useState<
+    {
+      value: string | null;
+      label: string;
+    }[]
+  >([]);
+
+  useEffect(() => {
+    if (!bookmarkUtilities || gotParentFolder) return;
+
+    const parentFoldersFromFunc = bookmarkUtilities.getParentFolders();
+
+    setParentFolders([
+      ...parentFoldersFromFunc.map((e) => ({
+        value: e.Id,
+        label: e.Name === "Root" ? "No Parent" : e.Name,
+      })),
+    ]);
+
+    setGotParentFolder(true);
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
Don't mute the linter. Add in the dependencies all the values used inside the effect

Reply:
exhaustive-deps is the one people use the most.. It just causes unecessary refreshes

Reply:
And if you miss the required rerender?